CMakeLists.txt

```diff
@@ -116,14 +116,68 @@ endif()
 
 # Add Rythe
 set(RYTHE_ENABLED_MODULE_NAMES "")
+set(RYTHE_ENABLED_APPLICATION_NAMES "")
 set(RYTHE_ENABLED_MODULE_PATHS "")
 add_subdirectory(${RYTHE_DIR_RYTHE})
 
 # The engine is ready - add applications
-if (${RYTHE_BUILD_APPLICATIONS})
+if ((${RYTHE_BUILD_APPLICATIONS}))
 	add_subdirectory(${RYTHE_DIR_ROOT}/applications)
 
 	foreach(path ${RYTHE_ENABLED_MODULE_PATHS})
 		add_subdirectory(${RYTHE_DIR_ROOT}/${path}/applications)
 	endforeach()
 endif()
+
+# Add config project
+set(configure_command_args "")
+list(APPEND configure_command_args "\"${RYTHE_DIR_ROOT}\"")
+list(APPEND configure_command_args "-G \"${CMAKE_GENERATOR}\"")
+list(APPEND configure_command_args "-S \"${RYTHE_DIR_ROOT}\"")
+list(APPEND configure_command_args "-B \"${CMAKE_BINARY_DIR}\"")
+list(APPEND configure_command_args -T ClangCL)
+list(APPEND configure_command_args -D RYTHE_FETCH_SUBMODULES=OFF)
+
+if ((${RYTHE_BUILD_APPLICATIONS}))
+	list(APPEND configure_command_args -D RYTHE_BUILD_APPLICATIONS=ON)
+else()
+	list(APPEND configure_command_args -D RYTHE_BUILD_APPLICATIONS=OFF)
+endif()
+
+if ((${RYTHE_BUILD_OPTION_ASAN}))
+	list(APPEND configure_command_args -D RYTHE_BUILD_OPTION_ASAN=ON)
+else()
+	list(APPEND configure_command_args -D RYTHE_BUILD_OPTION_ASAN=OFF)
+endif()
+
+if ((${RYTHE_BUILD_OPTION_WERROR}))
+	list(APPEND configure_command_args -D RYTHE_BUILD_OPTION_WERROR=ON)
+else()
+	list(APPEND configure_command_args -D RYTHE_BUILD_OPTION_WERROR=OFF)
+endif()
+
+if ((${RYTHE_FORCE_ENABLE_ALL_MODULES}))
+	list(APPEND configure_command_args -D RYTHE_FORCE_ENABLE_ALL_MODULES=ON)
+else()
+	list(APPEND configure_command_args -D RYTHE_FORCE_ENABLE_ALL_MODULES=OFF)
+	foreach(module_name ${RYTHE_ENABLED_MODULE_NAMES})
+		string(TOUPPER module_name module)
+		list(APPEND configure_command_args -D RYTHE_MODULE_${module}=ON)
+	endforeach()	
+endif()
+
+add_custom_target(configure ${CMAKE_COMMAND} ${configure_command_args})
+
+foreach(module_name ${RYTHE_ENABLED_MODULE_NAMES})
+	add_dependencies(${module_name} configure)
+endforeach()
+
+foreach(application_name ${RYTHE_ENABLED_APPLICATION_NAMES})
+	add_dependencies(${application_name} configure)
+endforeach()
+
+# FOLDER THINGS
```

Comment:
Change to a more descriptive comment like "Set the folder names. Use to organize targets in IDEs."